### PR TITLE
[ADD] Add base_phone_validation module.

### PR DESCRIPTION
**Pull request by foutoucour**
Module that complement base_phone by forcing the phone numbers to be validated against the standards of the partner's country

### CODE REVIEW

**Comment by foutoucour** [comment]
Hey @alexis-via 
I know you did a lot of work for base_phone and similar modules. Could give your 2cents on this module?

**Comment by alexis-via** [comment]
I will comment in more details later on. But I'm a bit surprised by your technical approach and I don't think we need a separate module for that.

In the base_phone module, you have a parameter "raise_if_phone_parse_fails" that is read from the context and should suit your needs. Or maybe we should add a configuration parameter in the "base_phone" module to set the default value of raise_if_phone_parse_fails.

File: base_phone_validation/README.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Base Phone Validation
+=====================
+Force phone numbers to be valid against the the country of the partner.
```
Comment (alexis-via) [inline base_phone_validation/README.rst:3]: double "the"

File: base_phone_validation/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#
+##############################################################################
+
+{
+    'name': 'Base Phone Validation',
+    'version': '1.0',
+    'author': 'cgstudiomap, Odoo Community Association (OCA)',
+    'maintainer': 'cgstudiomap',
+    'license': 'AGPL-3',
+    'category': 'Sale',
```
Comment (alexis-via) [inline base_phone_validation/__openerp__.py:28]: let's put the same category as the base_phone module

File: base_phone_validation/README.rst [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+Contributors
+------------
+* Jordi Riera <kender.jr@gmail.com>
+
```
Comment (alexis-via) [inline base_phone_validation/README.rst:8]: For the file README.rst, you should use the official OCA template available here: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

**Comment by alexis-via** [comment]
With my few remarks, I approve this PR: :+1:

**Comment by ghost** [comment]
LGTM :+1:

@foutoucour do you plan on fixing the lint checks? I would like to merge this PR and migrate it to 10.0.

**Comment by ventilooo** [comment]
@foutoucour @dufresnedavid  already ported to [10.0](https://github.com/savoirfairelinux/connector-telephony/tree/base_phone_validation_V10) #146

**Pull request closed without merging**